<?xml version="1.0"?>

<robot name="Puzzlebot Jetson-Lidar Ed.">

<!-- Dummy link to fiz robot chassis -->
<link name="base_link">
</link>

<link name="chassis">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="7.58333333333e-06" ixy="0.0" ixz="0.0" iyy="7.58333333333e-06" iyz="0.0" izz="1.25e-05"/>
        </inertial>

        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Chassis.stl"/>
            </geometry>
        </collision>

        <visual>
            <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Chassis.stl"/>
            </geometry>
        </visual>

</link>

<link name="wheel1">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="7.58333333333e-06" ixy="0.0" ixz="0.0" iyy="7.58333333333e-06" iyz="0.0" izz="1.25e-05"/>
        </inertial>

        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_sim/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
        </collision>

        <visual>
            <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_sim/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
        </visual>
</link>

<link name="wheel2">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="7.58333333333e-06" ixy="0.0" ixz="0.0" iyy="7.58333333333e-06" iyz="0.0" izz="1.25e-05"/>
        </inertial>

        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_sim/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
        </collision>

        <visual>
            <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_sim/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
        </visual>
</link>



<joint name="chassis_joint" type="fixed">
    <parent link="base_link"/>
    <child link="chassis"/>
    <origin xyz="0.0 0.0 0.05"/>
</joint>

<joint name="wheel_joint1" type="continuous">
    <parent link="chassis"/>
    <child link="wheel1"/>
    <origin xyz="0.0 -0.15 0.0" rpy="0 0 1.5708"/>
</joint>

<joint name="wheel_joint2" type="continuous">
    <parent link="chassis"/>
    <child link="wheel2"/>
    <origin xyz="0.0 0.15 0.0" rpy="0 0 -1.5708"/>
</joint>

<!-- FINISH THE URDF-->

</robot>
